src/ci/docker/host-x86_64/i686-gnu-nopt/Dockerfile

```diff
@@ -27,5 +27,8 @@ RUN echo "[rust]" > /config/nopt-std-config.toml
 RUN echo "optimize = false" >> /config/nopt-std-config.toml
 
 ENV RUST_CONFIGURE_ARGS --build=i686-unknown-linux-gnu --disable-optimize-tests
-ENV SCRIPT python3 ../x.py test --stage 0 --config /config/nopt-std-config.toml library/std \
+ARG SCRIPT_ARG
+ENV DEFAULT_SCRIPT python3 ../x.py test --stage 0 --config /config/nopt-std-config.toml library/std \
```

Comment:
I would just remove the default, and require the script to be set. It will make it slightly harder to run this image locally, but it is not possible to exactly reproduce the `-1` and `-2` variants from the YAML file anyway.

Side note: splitting the jobs without duplicating the Dockerfiles makes it harder to reproduce the CI jobs locally, because the source of truth is no longer in the Dockerfile. One of the goals of the future improvement of our CI setup should be to allow running the Docker runners based on source of truth in `jobs.yml`.